vsintegration/src/vs/FsPkgs/FSharp.LanguageService/servicem.fs

```diff
@@ -2016,12 +2021,32 @@ type FSharpPackage() as self =
     let callback = new ServiceCreatorCallback(CreateIfEnabled)
     
     let mutable mgr : IOleComponentManager = null
+
+    let fsharpSpecificProfileSettings =
+        [| "TextEditor.F#.Insert Tabs", box false
+           "TextEditor.F#.Brace Completion", box true
+           "TextEditor.F#.Indent Style", box 1u |]
     
     override self.Initialize() =
         UIThread.CaptureSynchronizationContext()
+        self.EstablishDefaultSettingsIfMissing()
         (self :> IServiceContainer).AddService(typeof<FSharpLanguageService>, callback, true)
         base.Initialize()
 
+    /// In case custom VS profile settings for F# are not applied, explicitly set them here.
+    /// e.g. 'keep tabs' is the text editor default, but F# requires 'insert spaces'.
+    /// We specify our customizations in the General profile for VS, but we have found that in some cases
+    /// those customizations are incorrectly ignored.
+    member private this.EstablishDefaultSettingsIfMissing() =
```

Comment:
extract as function like `ensureDefaultSettings :  SVsSettingsPersistenceManager -> (string * object)` (maybe in a specific module), and invoke with `fsharpSpecificProfileSettings  |> Array.iter (ensureDefaultSettings  settingsManager)`, you can also inline the field fsharpSpecificProfileSettings  too
This is a good function to be reused

Also pattern match for null of settingsManager, the [roslyn guys](https://github.com/dotnet/roslyn/blob/334885aed18184734509a9598d5af3ce8a2cba3b/src/VisualStudio/Core/Impl/Options/AbstractSettingsManagerOptionSerializer.cs#L41-L43) does it because a null value in package initialization crash the ide .

The tools -> options -> Text Editor is a good place for new options :smile: